discord/ext/commands/core.py

```diff
@@ -1602,21 +1799,21 @@ def has_role(item):
         The name or ID of the role to check.
     """
 
-    def predicate(ctx):
+    def predicate(ctx: Context) -> bool:
         if not isinstance(ctx.channel, discord.abc.GuildChannel):
             raise NoPrivateMessage()
 
         if isinstance(item, int):
-            role = discord.utils.get(ctx.author.roles, id=item)
+            role = discord.utils.get(ctx.author.roles, id=item)  # type: ignore
         else:
-            role = discord.utils.get(ctx.author.roles, name=item)
+            role = discord.utils.get(ctx.author.roles, name=item)  # type: ignore
```

Comment:
Type-checker's going to think `ctx.author` could still be a `User`, could solve by checking the `ctx.author` instance is a `Member` instead of checking the `channel` is a `GuildChannel`